jvector-base/src/main/java/io/github/jbellis/jvector/graph/disk/FeatureId.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jbellis.jvector.graph.disk;
+
+import io.github.jbellis.jvector.disk.RandomAccessReader;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * An enum representing the features that can be stored in an on-disk graph index.
+ * Order of this enum is critical! It will be serialized on-disk in enum order.
+ * These are typically mapped to a Feature or FeatureWriter.
+ */
+public enum FeatureId {
+    INLINE_VECTORS(InlineVectors::load),
```

Comment:
#collaborating

it might be cleaner to have it take (int bitflag, loader) to make the flags explicit instead of implicit wrt the order of the enum 

means you can always get the flag from the enum itself without having to count through values()